src/adapters/communities-db.ts

```diff
@@ -7,5 +19,123 @@ export function createCommunitiesDBComponent(
 
   const logger = logs.getLogger('communities-db-component')
 
-  return {}
+  return {
+    async getCommunity(id: string, userAddress: string) {
+      const query = SQL`
+        SELECT 
+          c.id,
+          c.name,
+          c.description,
+          c.owner_address as "ownerAddress",
+          COALESCE(cm.role, ${CommunityRole.None}) as role,
+          CASE WHEN c.private THEN 'private' ELSE 'public' END as privacy,
+          c.active
+        FROM communities c
+        LEFT JOIN community_members cm ON c.id = cm.community_id AND cm.member_address = ${normalizeAddress(userAddress)}
+        WHERE c.id = ${id} AND c.active = true`
+
+      const result = await pg.query<Community>(query)
+
+      if (result.rows.length === 0) {
+        throw new CommunityNotFoundError(`Community not found: ${id}`)
+      }
+
+      return result.rows[0]
+    },
+
+    async getCommunityMembersCount(communityId) {
+      const query = SQL`
+        SELECT COUNT(DISTINCT cm.member_address) 
+          FROM community_members cm
+          LEFT JOIN community_bans cb ON cm.member_address = cb.banned_address 
+              AND cb.community_id = cm.community_id
+              AND cb.active = true
+          WHERE cm.community_id = ${communityId}
+              AND cb.banned_address IS NULL
+              AND EXISTS (
+                  SELECT 1 
+                  FROM communities c 
+                  WHERE c.id = cm.community_id 
+                  AND c.active = true
+              )
+      `
+      const result = await pg.query<{ count: number }>(query)
+      return result.rows[0].count
+    },
+
+    async getCommunityPlaces(communityId: string) {
+      const query = SQL`
+        SELECT 
+          CASE 
+            WHEN place_type = 'parcel' THEN position
+            ELSE world_name
+          END as place
+        FROM community_places
+        WHERE community_id = ${communityId}`
+      const result = await pg.query<{ place: string }>(query)
+      return result.rows.map((row) => row.place)
+    },
+
+    async getCommunities(memberAddress: string, options?: { pagination?: Pagination }) {
+      const { pagination = { limit: 10, offset: 0 } } = options ?? {}
```

Comment:
should we set a default global limit to use in all queries that requires pagination?